### PR TITLE
Dashboard: Add ability to update default publisher logo, context menu for logos

### PR DESCRIPTION
**Pull request by BrittanyIRL**
## Summary
Adds a menu to uploaded publisher logos that allows a user to remove or set a logo as default. 

logo updates

## Relevant Technical Choices

- New context menu for publisher logos, following story pattern 
- Adds ability to update default publisher logo 

## To-do

- [x] Functionality to automatically set the default logo to the first available logo that is initially uploaded when user does not already have any logos uploaded. 
- [x] Fix tests that are broken due to functionality changes (there's no longer a 'remove logo' button on each logo aside the default) 
- [ ] Clean up additions, there's probably room to refactor in here
- [x] keyboard functionality of new menu here will work better once keyboard accessibility PR for stories is merged. 

## User-facing changes

Now users get an 'edit' menu when they hover or focus on a publisher logo. Opening that menu, you get two actions - delete the logo or make it the default. 
If a user only has one logo and it's the default then it can't be removed and it (obviously) can't be made the default. 
If a user has no logos, the first one they upload (or the first one in a batch of uploads) becomes default. 

## Testing Instructions
- Update the default logo on the settings page
- Remove a logo on the settings page
- Remove a default logo on the settings page where user has more than one logo uploaded 
- You cannot make a default logo the default logo (menu item disabled) 
- If there's only one menu item, there is no menu. 

---

Addresses #3811

### CODE REVIEW

**Comment by BrittanyIRL** [comment]
@googlebot I consent

**Comment by swissspidy** [comment]
FYI @carlos-kelly you'd need to consent to the CLA bot as you made commits to someone else's PR.

**Comment by carloskelly13** [comment]
@googlebot I consent

File: assets/src/dashboard/app/views/editorSettings/karma/editorSettings.karma.js [hunk 1]
```diff
@@ -41,27 +41,27 @@ describe('Settings View', () => {
...
+  //   ) {
+  //     // eslint-disable-next-line no-await-in-loop
+  //     await fixture.events.keyboard.press('tab');
+  //     limit++;
+  //   }
+
+  //   return publisherLogosContainer.contains(document.activeElement)
+  //     ? Promise.resolve()
+  //     : Promise.reject(new Error('could not focus on publisher logos'));
+  // }
```
Comment (BrittanyIRL) [inline assets/src/dashboard/app/views/editorSettings/karma/editorSettings.karma.js]: Need PR #4317 merged first - it has the updated keyboard focus logic for this

File: assets/src/dashboard/app/views/editorSettings/publisherLogo/index.js [hunk 2]
```diff
@@ -80,30 +83,25 @@ function PublisherLogoSettings({
...
   );
 
-  // set active logo when first painting
-  useEffect(() => {
-    if (publisherLogos.length > 0 && !activePublisherLogo) {
-      setActivePublisherLogoId(publisherLogos?.[0].id);
-    }
-  }, [activePublisherLogo, publisherLogos, setActivePublisherLogoId]);
-
   // Update publisher logo focus when logo is removed
```
Comment (BrittanyIRL) [inline assets/src/dashboard/app/views/editorSettings/publisherLogo/index.js:104]: We don't want to do this now that there's hidden UI that shows up either when active or hovered upon

File: assets/src/dashboard/components/popoverMenu/menu.js [hunk 3]
```diff
@@ -186,7 +186,7 @@ const Menu = ({ isOpen, currentValueIndex = 0, items, onSelect }) => {
   );
 
   return (
-    <MenuContainer ref={listRef}>
+    <MenuContainer ref={listRef} {...rest}>
```
Comment (BrittanyIRL) [inline assets/src/dashboard/components/popoverMenu/menu.js]: this is for easier manipulation for test ids and referencing

**Review by dmmulroy — APPROVED** [review]

**Review by carloskelly13 — APPROVED** [review]
Works well including setting the first uploaded logo as default.

**Pull request merged**